src/components/checkbox/checkbox.js

```diff
@@ -171,12 +184,18 @@ function MdCheckboxDirective(inputDirective, $mdAria, $mdConstant, $mdTheming, $
       }
 
       function render() {
-        if(ngModelCtrl.$viewValue) {
+        if(ngModelCtrl.$viewValue && !isIndeterminant) {
           element.addClass(CHECKED_CSS);
         } else {
           element.removeClass(CHECKED_CSS);
         }
       }
+
+      function setIndeterminantState() {
+        isIndeterminant = scope.$eval(attr.mdIndeterminant) === false ? false : true;
+        element[0].indeterminate = isIndeterminant ? true : undefined;
```

Comment:
I believe this is to leverage some native browser implementation features. @ErinCoughlan?